samples/boards/sensortile_box_pro/ble-fw-upgrade/src/main.c

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * The BlueNRG-LP, BlueNRG-LPS UART bootloader protocol (AN5471)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/led.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ble_fw_upgrade, LOG_LEVEL_DBG);
+
+#define BLE_FLASH_START_ADDRESS 0x10040000
+
+/* Defined in ${ZEPHYR_HAL_ST_MODULE_DIR}/ble_firmware/ble_fw.c */
+extern const uint8_t ble_fw_img[];
+extern const int ble_fw_img_len;
+
+const struct gpio_dt_spec green_gpio = GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+const struct gpio_dt_spec red_gpio = GPIO_DT_SPEC_GET(DT_ALIAS(led1), gpios);
+const struct gpio_dt_spec blue_gpio = GPIO_DT_SPEC_GET(DT_ALIAS(led3), gpios);
+
+typedef enum {
+	BLE_FW_UPG_START = 0x0,
+	BLE_FW_UPG_OK = 0x1,
+	BLE_FW_UPG_ERR = 0x2,
+} ble_fw_upg_status;
+
+static int led_pattern_out(ble_fw_upg_status status)
```

Comment:
should this return void? and then simplify the caller code a bit

Reply:
yes, correct!